File: c_sharp/ElectionResult.cs
using System.Collections.Generic;

namespace c_sharp
{
	public class ElectionResult
	{
		public static ElectionResult Success(string content, string constraints, List<string> objections)
		{
			return new SuccessfulElectionResult(new Agreement(content, constraints), objections);
		}

		public static ElectionResult Failure(Proposal proposal, List<string> objections)
		{
			return new FailedElectionResult(proposal, objections);
		}

		public class SuccessfulElectionResult : ElectionResult
		{
			public SuccessfulElectionResult(Agreement agreement, List<string> objections)
			{
				Agreement = agreement;
				Objections = objections;
			}

			public Agreement Agreement { get; }
			public List<string> Objections { get; }
		}

		public class FailedElectionResult : ElectionResult
		{
			public FailedElectionResult(Proposal proposal, List<string> objections)
			{
				Proposal = proposal;
				Objections = objections;
			}

			public Proposal Proposal { get; }
			public List<string> Objections { get; }
		}
	}
}

File: LegacyPlay/ElectionResult.cpp
#include "pch.h"
#include "ElectionResult.h"

ElectionResult ElectionResult::Success(const std::string & content, const std::string & constraints, const std::vector<std::string>& objections)
{
	return ElectionResult();
}

ElectionResult ElectionResult::Failure(const Proposal & proposal, const std::vector<std::string>& objections)
{
	return ElectionResult();
}
